packaging/linux/systemd/keybase.gui.service

```
@@ -1,18 +1,28 @@
 [Unit]
 Description=Keybase GUI
+# Do not issue a hard dependency on service, because
+# gui can reconnect to a restarted service.
 Wants=keybase.service kbfs.service
 
+# Provides DISPLAY on modern systems
+BindsTo=graphical-session.target
+
 [Service]
 Type=simple
 ExecStart=/opt/keybase/Keybase
-# The environment file lets run_keybase pass along KEYBASE_START_UI. The
-# autostart file will set that to suppress showing the main window during boot.
-# %t is the XDG_RUNTIME_DIR, and the leading - means it's ok if the file is
-# missing.
-EnvironmentFile=-%t/keybase/keybase.gui.env
+
+# After starting, we clear the KEYBASE_AUTOSTART variable so the next time
+# systemctl --start keybase.gui is called, it will be maximized.
+# Autostart scripts must set KEYBASE_AUTOSTART=1 in the systemd
+# environment on every call in order to start the gui in a minimized state.
+# The official Linux packages use run_keybase, which imports the
+# environment variable into systemd during autostart.
+# We need to shell out because systemctl is sometimes in /bin and sometimes in
+# /usr/bin depending on the distro
+ExecStartPost=sh -c 'systemctl --user unset-environment KEYBASE_AUTOSTART'
```

```suggestion
ExecStartPost=/bin/sh -c 'systemctl --user unset-environment KEYBASE_AUTOSTART'
```

This needs to be an absolute path for the same reason you couldn't use systemctl here. :p

---

Good catch, thanks for the tip!

---

Odd, I was doing some testing and
```[Unit]
Description=myunit

[Service]
Type=simple
ExecStart=date
```
works on my Arch system (systemd 239) but fails due to absolute path error on a Ubuntu 16.04 docker (systemd 229). But the `env` trick works on both fine.